src/spetlr/delta/delta_handle.py

```diff
@@ -172,15 +213,15 @@ def upsert(
             return self.write_or_append(df, mode="append")
 
         temp_view_name = get_unique_tempview_name()
-        df.createOrReplaceTempView(temp_view_name)
+        df.createOrReplaceGlobalTempView(temp_view_name)
```

Comment:
Is this still need "df._jdf.sparkSession().sql(merge_sql_statement)" when having a global temp view?

Reply:
I believe so... ?

Reply:
If the handle upsert is used in a streaming context - the foreachbatch operation needs to have access to the global view.